### PR TITLE
BCF-991 json unmarshalling benchmarks

### PR DESCRIPTION
**Pull request by krehermann**
BFC-991 Faster json un/marshal for BlockHistoryEstimator

 currently hacked to submission for testing go/codec easyjson and std json
 
 had to employ all kinds of hacks to make them all work in the same tests. much clean needed once we finalize and approach
 
 ```

cltest random seed: 1677681123094144000
goos: darwin
goarch: arm64
pkg: github.com/smartcontractkit/chainlink/core/chains/evm/types
BenchmarkBlock_Small_JSONMarshal
BenchmarkBlock_Small_JSONMarshal-10                16707             71839 ns/op            4648 B/op         44 allocs/op
BenchmarkBlock_Small_CodecMarshal
BenchmarkBlock_Small_CodecMarshal-10               24181             50625 ns/op            7738 B/op         41 allocs/op
BenchmarkBlock_Small_EasyJSONMarshal
BenchmarkBlock_Small_EasyJSONMarshal-10            22554             52716 ns/op            4742 B/op         45 allocs/op
BenchmarkBlock_Medium_JSONMarshal
BenchmarkBlock_Medium_JSONMarshal-10                 736           1589566 ns/op          105132 B/op       1055 allocs/op
BenchmarkBlock_Medium_CodecMarshal
BenchmarkBlock_Medium_CodecMarshal-10                998           1189305 ns/op          180581 B/op       1056 allocs/op
BenchmarkBlock_Medium_EasyJSONMarshal
BenchmarkBlock_Medium_EasyJSONMarshal-10            1078           1094679 ns/op           98919 B/op       1027 allocs/op
BenchmarkBlock_Large_JSONMarshal
BenchmarkBlock_Large_JSONMarshal-10                   87          12914635 ns/op          842485 B/op       8343 allocs/op
BenchmarkBlock_Large_CodecMarshal
BenchmarkBlock_Large_CodecMarshal-10                 124           9626953 ns/op         1441388 B/op       8331 allocs/op
BenchmarkBlock_Large_EasyJSONMarshal
BenchmarkBlock_Large_EasyJSONMarshal-10              135           8704657 ns/op          739686 B/op       8088 allocs/op
BenchmarkBlock_Small_JSONUnmarshal
BenchmarkBlock_Small_JSONUnmarshal-10               7948            145673 ns/op            2024 B/op         44 allocs/op
BenchmarkBlock_Small_CodecUnmarshal
BenchmarkBlock_Small_CodecUnmarshal-10             10000            103004 ns/op            5456 B/op         55 allocs/op
BenchmarkBlock_Small_EasyJSONUnmarshal
BenchmarkBlock_Small_EasyJSONUnmarshal-10           9778            120728 ns/op            1415 B/op         33 allocs/op
BenchmarkBlock_Medium_JSONUnmarshal
BenchmarkBlock_Medium_JSONUnmarshal-10               361           3288897 ns/op           55240 B/op        866 allocs/op
BenchmarkBlock_Medium_CodecUnmarshal
BenchmarkBlock_Medium_CodecUnmarshal-10              536           2229515 ns/op          137017 B/op       1249 allocs/op
BenchmarkBlock_Medium_EasyJSONUnmarshal
BenchmarkBlock_Medium_EasyJSONUnmarshal-10           489           2493115 ns/op           35758 B/op        546 allocs/op
BenchmarkBlock_Large_JSONUnmarshal
BenchmarkBlock_Large_JSONUnmarshal-10                 44          26506207 ns/op          426691 B/op       6700 allocs/op
BenchmarkBlock_Large_CodecUnmarshal
BenchmarkBlock_Large_CodecUnmarshal-10                94          17885374 ns/op         1073838 B/op       9770 allocs/op
BenchmarkBlock_Large_EasyJSONUnmarshal
BenchmarkBlock_Large_EasyJSONUnmarshal-10             69          19288492 ns/op          269917 B/op       4143 allocs/op
PASS
ok      github.com/smartcontractkit/chainlink/core/chains/evm/types     28.306s
```

### CODE REVIEW

**Review by samsondav — DISMISSED** [review]

File: core/chains/evm/config/v2/block_history_benchmark_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			b.Fatalf("err %+v", err)
+		}
+		buf = buf[:0]
+	}
+}
+
+func BenchmarkBlockHistoryEstimator_Unmarshal_Codec(b *testing.B) {
+	//buf := make([]byte, 0, expectedSize)
+	var h codec.Handle = new(codec.JsonHandle)
+	//enc := codec.NewEncoderBytes(&buf, h)
```
Comment (samsondav) [inline core/chains/evm/config/v2/block_history_benchmark_test.go:90]: Why the commented code?

**Review by samsondav — COMMENTED** [review]
Is it possible to remove any fields from the json parsing to increase performance a bit more?

File: core/chains/evm/config/v2/block_history_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+			b.Fatalf("err %+v", err)
+		}
+		buf = buf[:0]
+	}
+}
+
+func BenchmarkBlockHistoryEstimator_Unmarshal_Codec(b *testing.B) {
+	var h codec.Handle = new(codec.JsonHandle)
+	var temp BlockHistoryEstimator
+	for i := 0; i < b.N; i++ {
```
Comment (jmank88) [inline core/chains/evm/config/v2/block_history_benchmark_test.go:90]: nit/ insignificant with large enough `N`, but you can reset the timer before looping to leave out any overhead from allocations/setup:
```suggestion
	b.ResetTimer()
	for i := 0; i < b.N; i++ {
```

File: GNUmakefile [hunk 3]
```diff
@@ -102,6 +102,15 @@ presubmit: ## Format go files and imports.
 mockery: $(mockery) ## Install mockery.
 	go install github.com/vektra/mockery/v2@v2.20.0
 
+.PHONY: codecgen
+codecgen: $(codecgen)
+	go install github.com/ugorji/go/codec/codecgen@v1.2.10
+
+.PHONY: run-codecgen
+run-codecgen: $(run-codecgen)
+	cd core/chains/evm/config/v2 && codecgen -o config_codecgen.go -r "BlockHistory.*" -j true config.go
```
Comment (jmank88) [inline GNUmakefile:111]: Would it make sense to embed this in source as a `//go:generate` command?

**Comment by krehermann** [comment]
moved to https://github.com/smartcontractkit/chainlink/pull/8644

**Pull request closed without merging**